examples/api/lib/material/input_chip/input_chip.1.dart

```diff
@@ -0,0 +1,434 @@
+      (T data) {
+        final Widget child = widget.chipBuilder(context, data);
+        if (_lastChipFocusNode != null) {
+          _lastChipFocusNode!.requestFocus();
+          return Focus(focusNode: _lastChipFocusNode, child: child);
+        }
+        return child;
+      },
+    ).toList();
+
+    chipsChildren.add(Row(
+      mainAxisSize: MainAxisSize.min,
+      mainAxisAlignment: MainAxisAlignment.end,
+      crossAxisAlignment: CrossAxisAlignment.end,
+      children: <Widget>[
+        Text(text),
+        TextCaret(
+          resumed: _focusNode.hasFocus,
+        ),
+      ],
+    ));
+
+    return GestureDetector(
+      onTap: _requestKeyboard,
+      child: InputDecorator(
+        decoration: widget.decoration,
+        isFocused: _focusNode.hasFocus,
+        isEmpty: _value.text.isEmpty,
+        textAlignVertical: TextAlignVertical.center,
+        child: Wrap(
+          crossAxisAlignment: WrapCrossAlignment.center,
+          runSpacing: 4,
+          spacing: 4,
+          children: chipsChildren,
+        ),
+      ),
+    );
+  }
+
+  String get text {
+    return String.fromCharCodes(
+      _value.text.codeUnits.where((int ch) => ch != kObjectReplacementChar),
+    );
+  }
+
+  bool get _hasInputConnection => _connection?.attached ?? false;
+
+  void _requestKeyboard() {
+    if (_focusNode.hasFocus) {
+      _openInputConnection();
+    } else {
+      FocusScope.of(context).requestFocus(_focusNode);
+    }
+  }
+
+  void _onFocusChanged() {
+    if (_focusNode.hasFocus) {
+      _openInputConnection();
+    } else {
+      _closeInputConnectionIfNeeded();
+    }
+  }
+
+  void _openInputConnection() {
+    if (!_hasInputConnection) {
+      _connection = TextInput.attach(this, const TextInputConfiguration());
+      _connection!.setEditingState(_value);
+    }
+    _connection?.show();
+  }
+
+  void _closeInputConnectionIfNeeded() {
+    if (_hasInputConnection) {
+      _connection!.close();
+      _connection = null;
+    }
+  }
+
+  int _countReplacements(TextEditingValue value) {
+    return _countReplacementsInString(value.text);
+  }
+
+  int _countReplacementsInString(String value) {
+    return value.codeUnits
+        .where((int ch) => ch == kObjectReplacementChar)
+        .length;
+  }
+
+  int get currentReplacementCount => _countReplacements(_value);
+
+  void _syncTextValue() {
+    final TextEditingValue currentTextEditingValue = _value;
+    final int newChipCount = widget.values.length;
+    final int oldChipCount = _countReplacements(currentTextEditingValue);
+
+    String text = String.fromCharCodes(
+        List<int>.filled(newChipCount, kObjectReplacementChar));
+    if (newChipCount == oldChipCount) {
+      text += this.text;
+    }
+    _value = TextEditingValue(
+      text: text,
+      selection: TextSelection.collapsed(offset: text.length),
+    );
+    _connection?.setEditingState(_value);
+  }
+}
+
+class TextCaret extends StatefulWidget {
```

Comment:
This seems unnecessarily unconventional.

The state class is (conventionally):
```dart
class _TextCaretState extends State<TextCaret>
```
and so the TextCaret override would be:
```dart
State<TextCaret> createState() => _TextCaretState();
```